router/router_test.go

```diff
@@ -122,6 +122,8 @@ func Test_handleHTTPServiceRequest(t *testing.T) {
 	}
 }
 
+// TODO_TECHDEBT: replace all mockery autogenerated mocks with mockgen and move to own file
```

Comment:
```suggestion
// TODO_TECHDEBT(@commoddity): replace all mockery autogenerated mocks with mockgen and move to own file
```

Reply:
This is already done in a subsequent PR which should be merged soon.

Reply:
I've closed the router PR since the Envoy approach replaces the changes required in it so I've just made the router change in this branch rather than leaving a TODO.